activerecord/test/cases/relation/field_ordered_values_test.rb

```diff
@@ -57,4 +57,12 @@ def test_in_order_of_after_regular_order
 
     assert_equal(order, posts.map(&:id))
   end
+
+  def test_in_order_of_with_out_of_bound_integer
+    max_integer = Post.type_for_attribute(:id).send(:max_value)
+    order = [max_integer, 1]
+    posts = Post.where(type: "Post").order(:type).in_order_of(:id, order)
```

Comment:
I would go with this more visual example taken from your issue report:

```suggestion
    posts = Post.where(type: "Post").order(:type).in_order_of(:id, [1, 9999999999999])
```